k-distribution/include/kframework/builtin/domains.md

```diff
@@ -610,6 +610,72 @@ that two different sets will return the same element, even if they are similar.
 
 ```k
 endmodule
+
+module SET-KORE-SYMBOLIC [kore,symbolic]
+  imports SET
+  imports private K-EQUAL
+  imports private BOOL
+
+  // Symbolic in
+  //Unsupported: "Set1 Set2" matching. Using "Set SetItem(E) instead."
+  //rule E  in ( S1 S2)                  => E in S2 requires notBool E in S1       [simplification]
+  //rule E  in (S1 _S2)                  => true    requires         E in S1       [simplification]
+  rule E1 in (S SetItem(E2))           => true requires E1 ==K E2 orBool E1 in S     [simplification]
+  rule E1 in (S SetItem(E2))           => E1 in S           requires E1 =/=K E2      [simplification]                  
+  rule E1 in (S SetItem(E2))           => E1 in SetItem(E2) requires notBool E1 in S [simplification]
+
+  // Symbolic intersectSet
+  rule intersectSet(_S, .Set) => .Set                                                                                      [simplification]                                                                                         
+  rule intersectSet(.Set, _S) => .Set                                                                                      [simplification]
+  rule intersectSet(S, S) => S                                                                                             [simplification]
+  rule intersectSet(S1 SetItem(E), S2) => intersectSet(S1, S2) SetItem(E) requires         E in S2                         [simplification]
+  rule intersectSet(S1 SetItem(E), S2) => intersectSet(S1, S2)            requires notBool E in S2 ensures notBool E in S1 [simplification]     
+  rule intersectSet(S1, S2 SetItem(E)) => intersectSet(S1, S2) SetItem(E) requires         E in S1                         [simplification]
+  rule intersectSet(S1, S2 SetItem(E)) => intersectSet(S1, S2)            requires notBool E in S1 ensures notBool E in S2 [simplification]
```

Comment:
Same here, do we have an idea of the performance characteristics here? What if you are intersecting 2 100-element sets that are exactly duplicates of each other? What about 2 100-element sets that have no elements in common?

Reply:
One way around the non-determinism is to use `Set2List`, and then do list intersection, and then convert back to a `Set`. This may be the way to go.

Can we add tests of the above two scenarios? (the 100-element sets).